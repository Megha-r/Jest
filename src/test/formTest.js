
import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ConnectedForm from '../components/Form';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });
let wrapper;
let wrapperS;
const initialState = {}; 
const mockStore = configureStore();
let store;
describe('Form Component', () => {
 
  beforeEach(()=>{
    wrapper = shallow(<ConnectedForm />);
  })
  
  it('renders correctly', () => {
    const tree = renderer.create(
      <ConnectedForm />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });


  it('should render without throwing an error', () => {
      expect(wrapper.find('form.listing').exists()).toBe(true)
    })
   })

   it('renders a text input', () => {
    expect(wrapper.find('#textInput').length).toEqual(1)
   })

   describe('On Submit Function', () => {
    it('`<form>` element should have a onSubmit attribute', () => {
      expect(
        wrapper.props().onSubmit
      ).toBeDefined();
    });
  })
  
   
   describe('Text input', () => { 

    beforeEach(() => {
      store = mockStore(initialState)
      wrapperS = shallow(<ConnectedForm store={store}/>)
     
      }) 
    // it('should respond to change event and change the state of the Input Component', () => {
    // //  console.log(wrapperS.find('#title').at(0))
    //  wrapperS.find('#title').at(0).simulate('change', { value: 'React with Redux'});
     
    // //  const input = wrapperS.find('#title').get(0);
    // //  input.value = 'React with Redux';
    // expect(wrapperS.state('title')).toEqual('React with Redux');

    // }) 
    
  
   it('should call handleChange function', () => {
    const handleChange = jest.fn();
  
    const component = shallow(<ConnectedForm onChange={handleChange} />);
    const input = component.find('#textInput');

    input.simulate('props.onChange',{target: {value: 'title'}});
    expect(handleChange).toHaveReturnedTimes(0);
    });


    describe('<Save Button>', () => { 
      const testValues = {
        title: 'Hello',
        handleSubmit: jest.fn(),
    };
    it('Submit works', () => {

      const component = shallow(
          <ConnectedForm {...testValues} />
      );
      component.find('#savedBtn').simulate('click');
      expect(testValues.handleSubmit).toHaveBeenCalledTimes(0);
  });
});

})




