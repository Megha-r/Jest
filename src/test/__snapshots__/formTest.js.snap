// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component renders correctly 1`] = `
<form
  className="listing"
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      htmlFor="title"
    >
      Title
    </label>
    <input
      className="form-control form-control"
      id="textInput"
      onChange={[Function]}
      placeholder="Enter any string"
      type="text"
      value=""
    />
  </div>
  <button
    className="btn btn-success btn-lg btn btn-info"
    id="savedBtn"
    onClick={[Function]}
    type="submit"
  >
    SAVE
  </button>
</form>
`;
